spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -329,8 +335,10 @@ public boolean enableBatchRead() {
 
       boolean batchReadsEnabled = batchReadsEnabled(allParquetFileScanTasks, allOrcFileScanTasks);
 
+      boolean hasTimestampWithoutZone = SparkUtil.hasTimestampWithoutZone(lazySchema());
+
       this.readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && (allOrcFileScanTasks ||
-          (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
+              (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives && !hasTimestampWithoutZone));
```

Comment:
wasI think I noted on the other PR I don't think we need this

Reply:
This code was added by @bkahloon, but I agree that we don't need this check.
I have removed it